pkg/controller/dataimportcron-controller.go

```diff
@@ -719,6 +712,7 @@ func (r *DataImportCronReconciler) handleSnapshot(ctx context.Context, dataImpor
 		},
 	}
 	r.setDataImportCronResourceLabels(dataImportCron, desiredSnapshot)
+	cc.CopyAllowedLabels(pvc.GetLabels(), desiredSnapshot, false)
```

Comment:
So this is already great, but I think to be truly backwards compatible (user carries a snapshot object over upgrade) we should also update an existing snapshot with the label. Something like https://github.com/kubevirt/containerized-data-importer/pull/3155/commits/755fe0bc6a612b55c426b3186a51ea12e5274ce2#diff-786a9fcaa47bbf5a42fae757e2e41d06f646e2e8b778e9c17a6ab595ef9a52fbR360-R373
Maybe this is why the functional test fails? not sure. Have to take a deeper look, very odd

Reply:
It should not be needed since in this case the labels should remain on an existing `DataSource`. There would also be no way to regain this information without reimporting again.